command/build.go

```diff
@@ -171,6 +181,29 @@ func (c *BuildCommand) RunContext(buildCtx context.Context, cla *BuildArgs) int
 		c.Ui.Say("Debug mode enabled. Builds will not be parallelized.")
 	}
 
+	// TODO find an option that is not managed by a globally shared Publisher.
+	// This build currently enforces a 1:1 mapping that one publisher can be assigned to a single packer config file.
+	// It also requires that each config type implements this ConfiguredArtifactMetadataPublisher to return a configured bucket.
```

Comment:
I already expressed that option during our last zoom, but I think that we should handle that on a build block here:

https://github.com/hashicorp/packer/blob/9d0193ac14fb332f414e05b8755673d13b67fc56/hcl2template/types.build.go#L42-L51

If a build block contains a HCP setting, then all builds from that block will be push to HCP, makes it easy to read what interacts with HCP from the config file POW.

Internally, though, I would make each `builder` or `build` do it own thing independently of other builds.